### PR TITLE
Add create seed/example project and generate ui schema

### PR DESCRIPTION
**Pull request by TheZoker**
- Add `create seed project` function to theia and vscode
- Add `create example project` function to theia and vscode
- Add `generate UI schema` function to theia and vscode

### CODE REVIEW

File: theia-plugin/src/extension.ts [hunk 1]
```diff
@@ -1,21 +1,174 @@
...
-    }));
+export const start = (context: theia.PluginContext) => {
+  const createExampleProjectCommand = {
+    id: 'create-example-project',
+    label: 'JSONForms: Create Example Project',
+  };
+  const createExampleProject = theia.commands.registerCommand(
+    createExampleProjectCommand,
+    (args: any) => {
+    if (!args) {
```
Comment (edgarmueller) [inline theia-plugin/src/extension.ts]: Formatting/Indentation seems off in all commands

File: vscode-extension/src/extension.ts [hunk 2]
```diff
@@ -1,29 +1,167 @@
...
+export const activate = (context: vscode.ExtensionContext) => {
 
-    // Use the console to output diagnostic information (console.log) and errors (console.error)
-    // This line of code will only be executed once when your extension is activated
-    console.log('Congratulations, your extension "vscode-extension" is now active!');
+  // The command has been defined in the package.json file
+  // Now provide the implementation of the command with  registerCommand
+  // The commandId parameter must match the command field in package.json
+  const createExampleProject = vscode.commands.registerCommand(
+    'extension.createExampleProject',
```
Comment (edgarmueller) [inline vscode-extension/src/extension.ts]: Out of curiosity: are the 1st parameter and the `vscode` vs `theia` namespaces the only differences during command registration (we'll probably leave it as is in any case)?

**Comment by edgarmueller** [comment]
@eneufeld Please also take a look if you find the time.

**Comment by edgarmueller** [comment]
FYI: We'll squash the history, so you'll need to rebase.

File: theia-plugin/src/extension.ts [hunk 3]
```diff
@@ -1,21 +1,174 @@
+// tslint:disable:no-var-requires
+// tslint:disable:no-require-imports
+// tslint:disable:no-use-before-declare
 
 /**
  * Generated using theia-plugin-generator
  */
 
 import * as theia from '@theia/plugin';
+const tooling = require('jsonforms-tooling');
```
Comment (eneufeld) [inline theia-plugin/src/extension.ts]: can't we use an import instead of require here?

File: vscode-extension/src/extension.ts (hunk 2 above)
Comment (TheZoker) [reply vscode-extension/src/extension.ts]: No it's not the only difference. Also the data that is provided to the `registerCommand` function is different. But apart from these two differences, the code is identical.

File: theia-plugin/src/extension.ts (hunk 1 above)
Comment (TheZoker) [reply theia-plugin/src/extension.ts]: Fixed

File: theia-plugin/src/extension.ts (hunk 3 above)
Comment (TheZoker) [reply theia-plugin/src/extension.ts]: I replaced the `require` with `import` although it seems, that we have the  `Cannot find module 'jsonforms-tooling'` error again:
screenshot_2018 12 16_21h11m07s_010_

Maybe you could try the latest commit (run a `npm install` and a `npm run compile` inside the root folder) and tell me if you have the same issue or if it works fine for you

**Comment by edgarmueller** [comment]
@TheZoker Please create the issue as @eneufeld suggests (on this repo) and also move it to your internal backlog. Thanks!

**Comment by TheZoker** [comment]
@edgarmueller Although vscode uses npm and theia uses yarn?

**Comment by edgarmueller** [comment]
Yes, or are there any dependencies on npm/yarn in the code itself? I don't see any. As far as I see, you'd only need to extract the mentioned code blocks are parameterize them accordingly.

**Pull request merged**